moshi/src/test/java/com/squareup/moshi/ValueSourceEscapesTest.java

```diff
@@ -0,0 +1,68 @@
+package com.squareup.moshi;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import okio.Buffer;
+import okio.BufferedSource;
+import okio.Okio;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public final class ValueSourceEscapesTest {
+  @Test public void escapedDoubleQuote() throws IOException {
+    testSpecialCharacter("\\\"escaped\\\"");
+  }
+
+  @Ignore @Test public void escapedSingleQuote() throws IOException {
+    testSpecialCharacter("\\'escaped\\'");
+  }
+
+  @Test public void newLine1() throws IOException {
+    testSpecialCharacter("\\n");
+  }
+
+  @Test public void newLine2() throws IOException {
+    testSpecialCharacter("\\r\\n");
+  }
+
+  @Test public void stringTerminals() throws IOException {
+    testSpecialCharacter("{}[]:, \\n\\t\\r\\f/\\\\;#=");
+  }
+
+  @Ignore @Test public void backlash1() throws IOException {
+    testSpecialCharacter("\\/");
+  }
+
+  @Test public void backlash2() throws IOException {
+    testSpecialCharacter("\\\\");
+  }
+
+  private void testSpecialCharacter(String value) throws IOException {
+    String input = "{\"a\":\"" + value + "\"}";
+    JsonReader reader = JsonReader.of(Okio.buffer(Okio.source(new ByteArrayInputStream(input.getBytes()))));
+    Buffer out = new Buffer();
+    JsonWriter writer = JsonWriter.of(out);
+
+    reader.beginObject();
+    writer.beginObject();
```

Comment:
@swankjesse I wonder if this could ultimately be a solution to your previously mentioned desire to have a reader/writer symmetry where the write transforms the json going into the reader and can omit/modify certain keys/values